### PR TITLE
Add data binding proposal & update spec

### PR DESCRIPTION
**Pull request by dilanSachi**
## Purpose
$subject
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/2818
## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [] Added tests
- [x] Updated the spec

### CODE REVIEW

File: docs/spec/spec.md [hunk 1]
```diff
@@ -641,16 +704,18 @@ The negatively-acknowledged (rejected) messages can be re-queued by setting the
...
    }
    service rabbitmq:Service on channelListener {
-       remote function onRequest(rabbitmq:Message message) returns string {
-           string|error messageContent = string:fromBytes(message.content);
-           if messageContent is string {
-               log:printInfo("Received message: " + messageContent);
-           } 
+       remote function onRequest(Person person) returns string {
+           log:printInfo("Received person data: " + person.toString());
            return "Hello back from ballerina!";
```
Comment (shafreenAnfar) [inline docs/spec/spec.md:719]: I think we can do `return Hello + person.name;`

File: docs/spec/spec.md [hunk 2]
```diff
@@ -447,6 +447,44 @@ The `rabbitmq:Message` record received can be used to retrieve its contents.
...
+```
+If metadata like `routingKey`, `properties` are not needed, `content` can be directly received as well.
+```ballerina
+   rabbitmq:Service listenerService =
+   @rabbitmq:ServiceConfig {
+      queueName: "MyQueue"
+   }
+   service object {
+      remote function onRequest(string payload) returns string {
+         return "Hello Back!";
```
Comment (shafreenAnfar) [inline docs/spec/spec.md:484]: Maybe we can just echo back the payload.

File: docs/spec/spec.md [hunk 3]
```diff
@@ -447,6 +447,44 @@ The `rabbitmq:Message` record received can be used to retrieve its contents.
...
+       *rabbitmq:AnydataMessage;
+       string content;
+   |};
+   rabbitmq:Service listenerService =
+   @rabbitmq:ServiceConfig {
+      queueName: "MyQueue"
+   }
+   service object {
+      remote function onRequest(StringMessage message) returns string {
+         return "Hello Back!";
```
Comment (shafreenAnfar) [inline docs/spec/spec.md:462]: ditto

File: docs/proposals/data-binding.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+        string messageContent = check string:fromBytes(message.content);
+        Person person = check value:fromJsonStringWithType(messageContent);
+    }
+}
+```
+
+Receiving the message as a request,
+```ballerina
+service "MyQueue" on rabbitmqListener {
+    remote function onRequest(rabbitmq:Message message) returns anydata|rabbitmq:Error? {
```
Comment (shafreenAnfar) [inline docs/proposals/data-binding.md:53]: here we can do `string|rabbitmq:Error? `

Comment (dilanSachi) [reply docs/proposals/data-binding.md]: fixed in https://github.com/ballerina-platform/module-ballerinax-rabbitmq/pull/399/commits/3a0c50ad81de8fcbe2dab219b7971de22ca3be2e

File: docs/spec/spec.md (hunk 3 above)
Comment (dilanSachi) [reply docs/spec/spec.md]: fixed in https://github.com/ballerina-platform/module-ballerinax-rabbitmq/pull/399/commits/3a0c50ad81de8fcbe2dab219b7971de22ca3be2e

File: docs/spec/spec.md (hunk 2 above)
Comment (dilanSachi) [reply docs/spec/spec.md]: fixed in https://github.com/ballerina-platform/module-ballerinax-rabbitmq/pull/399/commits/3a0c50ad81de8fcbe2dab219b7971de22ca3be2e

File: docs/spec/spec.md (hunk 1 above)
Comment (dilanSachi) [reply docs/spec/spec.md]: fixed in https://github.com/ballerina-platform/module-ballerinax-rabbitmq/pull/399/commits/3a0c50ad81de8fcbe2dab219b7971de22ca3be2e

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**